expression/builtin_encryption_vec.go

```diff
@@ -139,11 +145,67 @@ func (b *builtinMD5Sig) vecEvalString(input *chunk.Chunk, result *chunk.Column)
 }
 
 func (b *builtinSHA2Sig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinSHA2Sig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalString(b.ctx, input, buf); err != nil {
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf1); err != nil {
+		return err
+	}
+	result.ReserveString(n)
+	i64s := buf1.Int64s()
+	var (
+		hasher    *hash.Hash
```

Comment:
Just use `hasher hash.Hash`, and it works.

Reply:
ok, I will fix it.